class-components-todo/src/App.js

```diff
@@ -0,0 +1,168 @@
+import React, { Component } from "react";
+import './App.css';
+import uuid from 'react-uuid';
+import { FaChevronDown } from "react-icons/fa";
+import  Button  from "@mui/material/Button";
+
+class App extends Component{
+  constructor(props) {
+    super(props);
+
+    // getInitialState
+    this.state = {
+      all: [],
+  newToDo:"",
+  mode:"all",
+  editToDo:"",
+  editToDoId:""
+    };
+  }
+
+  handleChange=(event)=>{
+    this.setState({newToDo:event.target.value});
+  };
+
+  handleKeypress = e => {
+    //it triggers by pressing the enter key
+  if (e.key==='Enter') {
+    this.newTodo();
+    this.setState({newToDo:""});
+  }
+};
+
+  newTodo=()=>{
+    this.setState({
+      all:[
+        ...this.state.all,
+        {id:uuid(),action:this.state.newToDo,done:false,isEdit:false}
+      ]});
+  };
+
+
+  toggleDone = (todo) =>
+  this.setState({
+    all: this.state.all.map((item) =>
+      item.action === todo.action ? { ...item, done: !item.done } : item
+    ),
+  });
+
+  delete=(todo)=>{
+    this.setState({ all: this.state.all.filter((item) => this.state.all.indexOf(item) !== this.state.all.indexOf(todo)) });
+  }
+
+  handleEdit= e =>{
+    this.setState({editToDo : e.target.value,editToDoId:e.target.id});
```

Comment:
is 'editTodoId' state really needed? You can just pass the ID in whichever function it is being used